clean-abap/CleanABAP_es.md

```diff
@@ -0,0 +1,5123 @@
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Tablas](#tablas) > [Esta sección](#prefiere-loop-at-where-a-if-anidado)
+
+```ABAP
+LOOP AT my_table REFERENCE INTO DATA(line) WHERE key = 'A'.
+```
+
+expresa la intención más claro y corto que
+
+```ABAP
+LOOP AT my_table REFERENCE INTO DATA(line).
+  IF line->key = 'A'.
+    EXIT.
+  ENDIF.
+ENDLOOP.
+```
+
+### Evita lecturas innecesarias a tablas
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Tablas](#tablas) > [Esta sección](#evita-lecturas-innecesarias-a-tablas)
+
+En caso de que _esperes_ que un registro esté ahí,
+lee una vez y reacciona a la excepción,
+
+```ABAP
+TRY.
+    DATA(row) = my_table[ key = input ].
+  CATCH cx_sy_itab_line_not_found.
+    RAISE EXCEPTION NEW /clean/my_data_not_found( ).
+ENDTRY.
+```
+
+en lugar de contaminar y alentar el flujo de control
+principal con una lectura doble.
+
+```ABAP
+" anti-pattern
+IF NOT line_exists( my_table[ key = input ] ).
+  RAISE EXCEPTION NEW /clean/my_data_not_found( ).
+ENDIF.
+DATA(row) = my_table[ key = input ].
+```
+
+> Además de ser una mejora de rendimiento,
+> esta es una variante especial de la sección más general
+> [Enfócate en el happy path o en manejo de errores, no en ambos](#enfócate-en-el-happy-path-o-en-manejo-de-errores-no-en-ambos).
+
+## Strings
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Esta sección](#strings)
+
+### Usa ` para definir literales
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Strings](#strings) > [Esta sección](#usa--para-definir-literales)
+
+```ABAP
+CONSTANTS some_constant TYPE string VALUE `ABC`.
+DATA(some_string) = `ABC`.  " --> TYPE string
+```
+
+Evita usar `'`, ya que hace una conversión de tipo superflua y confunde al lector
+sobre si está lidiando con un `CHAR` o un `STRING`:
+
+```ABAP
+" anti-pattern
+DATA some_string TYPE string.
+some_string = 'ABC'.
+```
+
+`|` está generalmente bien, pero no puede ser usada para `CONSTANTS` y agrega overhead innecesario cuando se especifica un valor fijo:
+
+```ABAP
+" anti-pattern
+DATA(some_string) = |ABC|.
+```
+
+### Usa | para construir textos
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Strings](#strings) > [Esta sección](#usa--para-construir-textos)
+
+```ABAP
+DATA(message) = |Received HTTP code { status_code } with message { text }|.
+```
+
+Las plantillas o templates de string resaltan mejor qué es una literal
+y qué es una variable, especialmente si colocas múltiples variables en un texto.
+
+```ABAP
+" anti-pattern
+DATA(message) = `Received an unexpected HTTP ` && status_code && ` with message ` && text.
+```
+
+## Booleanos
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Esta sección](#booleanos)
+
+### Usa los booleanos sabiamente
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Booleanos](#booleanos) > [Esta sección](#usa-los-booleanos-sabiamente)
+
+Seguido encontramos casos donde los booleanos naturalmente parecen ser la opción
```

Comment:
```suggestion
A continuación, encontramos casos donde los booleanos naturalmente parecen ser la opción
```

Reply:
That's not what _seguido_ meant. New suggestion:
```suggestion
Frecuentemente encontramos casos donde los booleanos naturalmente parecen ser la opción
```